src/item.cpp

```diff
@@ -749,6 +839,60 @@ void Item::serializeAttr(PropWriteStream& propWriteStream) const
 		propWriteStream.write<uint8_t>(ATTR_SHOOTRANGE);
 		propWriteStream.write<uint8_t>(getIntAttr(ITEM_ATTRIBUTE_SHOOTRANGE));
 	}
+
+	if (hasCustomAttributes()) {
+		propWriteStream.write<uint8_t>(ATTR_CUSTOM_ATTRIBUTES);
+		propWriteStream.write<size_t>(attributes->getCustomAttributeMap().size());
+		for (const auto &entry : attributes->getCustomAttributeMap()) {
+			// Serializing key type and value
+			customAttrTypes keyType = entry.first.type;
+			propWriteStream.write<customAttrTypes>(keyType);
+			switch (keyType) {
+				case ATTR_STRING_TYPE: {
+					propWriteStream.writeString(boost::get<std::string>(entry.first.value));
+					break;
+				}
+
+				case ATTR_INTEGER_TYPE: {
+					propWriteStream.write<int64_t>(boost::get<int64_t>(entry.first.value));
+					break;
+				}
+
+				default: {
+					break;
+				}
+			}
+
+			// Serializing value type and value
+			customAttrTypes valueType = entry.second.type;
+			propWriteStream.write<customAttrTypes>(valueType);
+			switch (valueType) {
+				case ATTR_STRING_TYPE: {
+					propWriteStream.writeString(boost::get<std::string>(entry.second.value));
+					break;
+				}
+
+				case ATTR_INTEGER_TYPE: {
+					propWriteStream.write<int64_t>(boost::get<int64_t>(entry.second.value));
```

Comment:
I think that specifying the type in write is a little redundant:
```
propWriteStream.write(boost::get<int64_t>(entry.second.value));
```
is sufficient IMO.